docs/source/guides/data.rst

```diff
@@ -0,0 +1,380 @@
+parameter can be used in conjunction with any of the above use cases. You can choose to pass
+the batches sequentially or simultaneously, as is done for the training step.
+The default mode for validation and test DataLoaders is sequential.
+
+See the following for more details for the default sequential option:
+
+- :meth:`~pytorch_lightning.core.datamodule.LightningDataModule.val_dataloader`
+- :meth:`~pytorch_lightning.core.datamodule.LightningDataModule.test_dataloader`
+
+.. testcode::
+
+    def val_dataloader(self):
+        loader_1 = DataLoader()
+        loader_2 = DataLoader()
+        return [loader_1, loader_2]
+
+To combine batches of multiple test and validation DataLoaders simultaneously, one
+needs to wrap the DataLoaders with `CombinedLoader`.
+
+.. testcode::
+
+    from pytorch_lightning.trainer.supporters import CombinedLoader
+
+    def val_dataloader(self):
+        loader_1 = DataLoader()
+        loader_2 = DataLoader()
+        loaders = {'a': loader_a,'b': loader_b}
+        combined_loaders = CombinedLoader(loaders, "max_size_cycle")
+        return combined_loaders
+
+
+Test with additional data loaders
+=================================
+You can run inference on a test set even if the :func:`~pytorch_lightning.core.Lightning.LightningModule.test_dataloader` method hasn't been
+defined within your :class:`~pytorch_lightning.core.Lightning.LightningModule` instance. For example, rhis would be the case if your test data
+set is not available at the time your model was declared. Simply pass the test set to the :func:`~pytorch_lightning.trainer.trainer.Trainer.test` method:
+
+.. code-block:: python
+
+    # setup your data loader
+    test = DataLoader(...)
+
+    # test (pass in the loader)
+    trainer.test(test_dataloaders=test)
+
+--------------
+
+
+.. _sequences:
+
+
+***************
+Sequential Data
+***************
+
+Lightning has built in support for dealing with sequential data.
+
+
+Packed sequences as inputs
+==========================
+When using PackedSequence, do 2 things:
+
+1. Return either a padded tensor in dataset or a list of variable length tensors in the DataLoader collate_fn (example shows the list implementation).
+2. Pack the sequence in forward or training and validation steps depending on use case.
+
+.. testcode::
+
+    # For use in DataLoader
+    def collate_fn(batch):
+        x = [item[0] for item in batch]
+        y = [item[1] for item in batch]
+        return x, y
+
+    # In module
+    def training_step(self, batch, batch_nb):
+        x = rnn.pack_sequence(batch[0], enforce_sorted=False)
+        y = rnn.pack_sequence(batch[1], enforce_sorted=False)
+
+----------
+
+Truncated Backpropagation Through Time
+======================================
+There are times when multiple backwards passes are needed for each batch.
+For example, it may save memory to use Truncated Backpropagation Through Time when training RNNs.
+
+Lightning can handle TBTT automatically via this flag.
```

Comment:
We use tbptt in `tbptt_split_batch`, would be good to be consistent with the acronym:
```suggestion
Lightning can handle TBPTT automatically via this flag.
```

Reply:
Done !